### PR TITLE
feature: Add huggingface-llm 0.6.0 dlc images

### PR DESCRIPTION
**Pull request by xyang16**
*Issue #, if available:*

*Description of changes:*

Huggingface is going to release Text Generation Inference DLC. This PR is to add the dlc images.

*Testing done:*

Unit test

`tests/unit/sagemaker/image_uris/test_huggingface_llm.py`

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/huggingface/llm_utils.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        )
+
+
+def get_huggingface_llm_image_uri(
+    region: Optional[str] = None,
+    version: Optional[str] = None,
+    py_version: Optional[str] = None,
+    instance_type: Optional[str] = None,
+    container_version: Optional[str]=None,
+    pytorch_version: Optional[str] = None,
```
Comment (lanking520) [inline src/sagemaker/huggingface/llm_utils.py:50]: why we need pytorch and tensorflow version here?

File: src/sagemaker/huggingface/llm_utils.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        raise ValueError(
+            "tensorflow_version and pytorch_version are both None. "
+            "Specify either tensorflow_version or pytorch_version."
+        )
+
+
+def get_huggingface_llm_image_uri(
+    region: Optional[str] = None,
+    version: Optional[str] = None,
+    py_version: Optional[str] = None,
```
Comment (lanking520) [inline src/sagemaker/huggingface/llm_utils.py:47]: why we need python version?

File: src/sagemaker/huggingface/llm_utils.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    if tensorflow_version is not None and pytorch_version is not None:
+        raise ValueError(
+            "tensorflow_version and pytorch_version are both not None. "
+            "Specify only tensorflow_version or pytorch_version."
+        )
+    if tensorflow_version is None and pytorch_version is None:
+        raise ValueError(
+            "tensorflow_version and pytorch_version are both None. "
+            "Specify either tensorflow_version or pytorch_version."
+        )
```
Comment (maaquib) [inline src/sagemaker/huggingface/llm_utils.py:41]:
```suggestion
    if (tensorflow_version is None) ^ (pytorch_version is None):
        raise ValueError(
            "Only one of tensorflow_version and pytorch_version must be specified."
        )
```

File: src/sagemaker/huggingface/llm_utils.py (hunk 1 above)
Comment (xyang16) [reply src/sagemaker/huggingface/llm_utils.py]: Removed.

File: src/sagemaker/huggingface/llm_utils.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+
+def get_huggingface_llm_image_uri(
```
[the remark this answers is not in the document]
Comment (xyang16) [reply src/sagemaker/huggingface/llm_utils.py:20]: Added.

File: src/sagemaker/huggingface/llm_utils.py (hunk 2 above)
Comment (xyang16) [reply src/sagemaker/huggingface/llm_utils.py]: Removed

**Review by philschmid — APPROVED** [review]
LGTM! Thats great work!

**Review by knikure — COMMENTED** [review]
/bot run all

**Comment by knikure** [comment]
@xyang16 Can you fix the black-check errors by running `tox -e black-format` command?
Submit your changes after successfully running `./.githooks/pre-push` command.

**Comment by xyang16** [comment]
> @xyang16 Can you fix the black-check errors by running `tox -e black-format` command? Submit your changes after successfully running `./.githooks/pre-push` command.

Done

**Review by knikure — COMMENTED** [review]
/bot run all

**Review by knikure — COMMENTED** [review]
/bot run pr

**Review by knikure — COMMENTED** [review]
/bot run all

**Comment by knikure** [comment]
@@xyang16  Can you take a look at failing tests?
```
tests/integ/test_training_compiler.py::test_pytorch[1.13.1-ml.p3.2xlarge-1]
```

**Comment by xyang16** [comment]
> @xyang16 Can you take a look at failing tests?
> 
> ```
> tests/integ/test_training_compiler.py::test_pytorch[1.13.1-ml.p3.2xlarge-1]
> ```

Could you please paste the specific error because I can't see what's failed?

I have tried to run this test myself, got error `cannot import name 'PartialState' from 'accelerate'"`. But my code didn't touch anything related to training. Maybe the failure is not related my PR.

**Review by knikure — COMMENTED** [review]
/bot run all

**Comment by xyang16** [comment]
@knikure Could you please take a look at the PR and run tests? Thanks

**Review by claytonparnell — COMMENTED** [review]
/bot run all

**Review by lanking520 — APPROVED** [review]

**Review by claytonparnell — COMMENTED** [review]
/bot run slow-tests

**Comment by ksachdeva11** [comment]
Hello,

I am getting the below error when trying the below code -  

```
from sagemaker.huggingface import get_huggingface_llm_image_uri

# retrieve the llm image uri
llm_image = get_huggingface_llm_image_uri(
  "huggingface",
  version="0.6.0"
)

# print ecr image uri
print(f"llm image uri: {llm_image}")
```

```
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In[55], line 1
----> 1 from sagemaker.huggingface import get_huggingface_llm_image_uri
      3 # retrieve the llm image uri
      4 llm_image = get_huggingface_llm_image_uri(
      5   "huggingface",
      6   version="0.6.0"
      7 )

ImportError: cannot import name 'get_huggingface_llm_image_uri' from 'sagemaker.huggingface' (/home/ec2-user/anaconda3/envs/python3/lib/python3.10/site-packages/sagemaker/huggingface/__init__.py)
```

I have already upgraded sagemaker sdk using this pip install git+https://github.com/xyang16/sagemaker-python-sdk.git@hf --upgrade

I am following the steps mentioned in this blog - https://huggingface.co/blog/sagemaker-huggingface-llm

Any help on this would be appreciated.

Thank you!

**Comment by ksachdeva11** [comment]
Using `from sagemaker.huggingface.llm_utils import get_huggingface_llm_image_uri` worked.

**Comment by xyang16** [comment]
@ksachdeva11 Please use `pip install sagemaker==2.161.0`. It should work.

**Comment by ksachdeva11** [comment]
@xyang16 Using `from sagemaker.huggingface.llm_utils import get_huggingface_llm_image_uri` worked.

**Comment by xyang16** [comment]
@ksachdeva11 If you install using `pip install sagemaker==2.161.0`, does `from sagemaker.huggingface import get_huggingface_llm_image_uri` work?

**Comment by xyang16** [comment]
@ksachdeva11 Maybe you could try `pip uninstall sagemaker` first, to make the environment clean.

**Pull request merged**